tests/integration/ha_tests/test_ha.py

```diff
@@ -65,6 +83,128 @@ async def test_build_and_deploy(ops_test: OpsTest) -> None:
+    ), "replica set primary is not one of the available units"
+
+    # verify that the configuration of mongodb no longer has the deleted ip
+    member_hosts = await fetch_replica_set_members(ops_test)
+
+    # verify that the replica set members have the correct units
+    assert set(member_hosts) == set(hostnames), "mongod config contains deleted units"
+
+    # verify that the no writes were skipped
+    total_expected_writes = await get_total_writes(ops_test)
+    with await get_mongo_client(ops_test) as client:
+        actual_writes = client[TEST_DB][TEST_COLLECTION].count_documents({})
+    assert total_expected_writes == actual_writes, "writes to the db were missed."
+
+
+async def test_replication_across_members(ops_test: OpsTest, continuous_writes) -> None:
+    """Check consistency, ie write to primary, read data from secondaries."""
+    # first find primary, write to primary, then read from each unit
+    await insert_focal_to_cluster(ops_test)
+
+    primary = await get_replica_set_primary(ops_test)
+    app = await get_application_name(ops_test, APP_NAME)
+    excluded = [primary]
+    for unit in ops_test.model.applications[app].units:
+        if unit.name == primary:
+            continue
+        with await get_mongo_client(ops_test, excluded=excluded) as client:
+            db = client["new-db"]
+            test_collection = db["test_ubuntu_collection"]
+            query = test_collection.find({}, {"release_name": 1})
+            assert query[0]["release_name"] == "Focal Fossa"
+        excluded.append(unit.name)
+
+    # verify that the no writes were skipped
+    total_expected_writes = await get_total_writes(ops_test)
+    with await get_mongo_client(ops_test) as client:
+        actual_writes = client[TEST_DB][TEST_COLLECTION].count_documents({})
+    assert total_expected_writes == actual_writes
+
+
+async def test_replication_member_scaling(ops_test: OpsTest, continuous_writes) -> None:
+    """Verify newly added and newly removed members properly replica data.
+
+    Verify newly members have replicated data and newly removed members are gone without data.
+    """
+    # first find primary, write to primary,
+    await insert_focal_to_cluster(ops_test)
+
+    app = await get_application_name(ops_test, APP_NAME)
+    original_units = [unit.name for unit in ops_test.model.applications[app].units]
+
+    await scale_application(ops_test, app, len(original_units) + 1)
+
+    with await get_mongo_client(ops_test, excluded=original_units) as client:
+        db = client["new-db"]
+        test_collection = db["test_ubuntu_collection"]
+        query = test_collection.find({}, {"release_name": 1})
+        assert query[0]["release_name"] == "Focal Fossa"
+
+        # verify that the no writes were skipped
+        total_expected_writes = await get_total_writes(ops_test)
+        actual_writes = client[TEST_DB][TEST_COLLECTION].count_documents({})
+        assert total_expected_writes == actual_writes
```

Comment:
Should the extra unit be scaled down once the test is done?

Reply:
see above comment about refactoring other tests.